packages/react/src/components/Tree/Tree.tsx

```diff
@@ -60,14 +83,53 @@ class Tree extends UIComponent<ReactProps<TreeProps>> {
     accessibility: defaultBehavior,
   }
 
+  static autoControlledProps = ['activeIndex']
+
+  getInitialAutoControlledState({ exclusive }): TreeState {
+    return {
+      activeIndex: exclusive ? -1 : [],
+    }
+  }
+
+  getActiveIndexAsArray() {
+    const { activeIndex } = this.state
+    return _.isArray(activeIndex) ? activeIndex : [activeIndex]
+  }
+
+  computeNewIndex = (index: number) => {
+    const { exclusive } = this.props
+
+    if (exclusive) return index
+    const activeIndexAsArray = this.getActiveIndexAsArray()
+    // check to see if index is in array, and remove it, if not then add it
+    return _.includes(activeIndexAsArray as number[], index)
+      ? _.without(activeIndexAsArray as number[], index)
+      : [...(activeIndexAsArray as number[]), index]
+  }
+
+  handleTreeItemOverrides = (predefinedProps: TreeItemProps) => ({
+    onTitleClick: (e: React.SyntheticEvent, treeItemProps: TreeItemProps) => {
+      this.trySetState({ activeIndex: this.computeNewIndex(treeItemProps.index) })
+      _.invoke(predefinedProps, 'onTitleClick', e, treeItemProps)
+    },
+  })
+
   renderContent() {
-    const { items, renderItemTitle } = this.props
+    const { items, renderItemTitle, exclusive } = this.props
+    const { activeIndex } = this.state
+    const activeIndexAsArray = this.getActiveIndexAsArray()
 
-    return _.map(items, item =>
+    return _.map(items, (item: ShorthandValue, index: number) =>
       TreeItem.create(item, {
         defaultProps: {
+          index,
+          exclusive,
           renderItemTitle,
+          open: exclusive
+            ? index === activeIndex
+            : _.includes(activeIndexAsArray as number[], index),
```

Comment:
```diff
-          open: exclusive
-            ? index === activeIndex
-            : _.includes(activeIndexAsArray as number[], index),
+         open  : _.includes(activeIndexes, index),

Reply:
For exclusive activeIndex is not an array. So this condition needs to be there.

Reply:
`this.getActiveIndexes()` will always return array, even with `exclusive=true`